### PR TITLE
branchprotector should not exit on 404

### PR DESCRIPTION
**Pull request by tehcyx**
For #16177, trying to provide a start for URL encode on branchprotector, as well as change the log.Fatalf to log.Errorf to not exit the process and force pod recreation.

edit: Change of logging is not necessary.

### CODE REVIEW

**Comment by tehcyx** [comment]
/cc @fejta @stevekuznetsov 
Would this be how we should deal with this? #16177

File: prow/cmd/branchprotector/protect.go [hunk 1]
```diff
@@ -132,7 +133,7 @@ func main() {
 		for i, err := range errors {
 			logrus.WithError(err).Error(i)
 		}
-		logrus.Fatalf("Encountered %d errors protecting branches", n)
+		logrus.Errorf("Encountered %d errors protecting branches", n)
```
Comment (petr-muller) [inline prow/cmd/branchprotector/protect.go]: I don't think this is a correct thing to do. Branchprotector is not a daemon, it executes and terminates. Exiting with non-zero is the right thing to do when something fails.

Comment (tehcyx) [reply prow/cmd/branchprotector/protect.go]: I see and guess I didn't read it too well. In that case we can leave the Fatalf for sure.

Comment (tehcyx) [reply prow/cmd/branchprotector/protect.go]: Reverted to Fatalf

**Pull request merged**